test/metabase/driver/query_processor/expand_resolve_test.clj

```diff
@@ -0,0 +1,286 @@
+(defn- obj->map [o]
+  (if-not (coll? o)
+    o
+    (cond
+      (or (vector? o)
+          (list? o))    (vec (for [v o]
+                               (obj->map v)))
+      (set? o)          (set (for [v o]
+                               (obj->map v)))
+      (map? o)          (into {} (for [[k v] o]
+                                   {k (obj->map v)})))))
+
+
+;; basic rows query w/ filter
+(expect
+  [;; expanded form
+   {:database (id)
+    :type     :query
+    :query    {:source-table (id :venues)
+               :filter       {:filter-type :>
+                              :field {:field-id (id :venues :price)
+                                      :fk-field-id nil
+                                      :datetime-unit nil}
+                              :value {:field-placeholder {:field-id (id :venues :price)
+                                                          :fk-field-id nil
+                                                          :datetime-unit nil}
+                                      :value 1}}}}
+   ;; resolved form
+   {:database {:name "test-data",
+              :details {:short-lived? nil, :db "mem:test-data;USER=GUEST;PASSWORD=guest"},
+              :id (id),
+              :engine :h2},
+   :type :query,
+   :query {:source-table {:schema "PUBLIC", :name "VENUES", :id (id :venues)},
+           :filter {:filter-type :>,
+                    :field {:field-id (id :venues :price),
+                            :field-name "PRICE",
+                            :field-display-name "Price",
+                            :base-type :IntegerField,
+                            :special-type :category,
+                            :visibility-type :normal,
+                            :table-id (id :venues),
+                            :schema-name "PUBLIC",
+                            :table-name "VENUES",
+                            :position nil,
+                            :description nil,
+                            :parent-id nil,
+                            :parent nil},
+                    :value {:value 1,
+                            :field {:field-id (id :venues :price),
+                                    :field-name "PRICE",
+                                    :field-display-name "Price",
+                                    :base-type :IntegerField,
+                                    :special-type :category,
+                                    :visibility-type :normal,
+                                    :table-id (id :venues),
+                                    :schema-name "PUBLIC",
+                                    :table-name "VENUES",
+                                    :position nil,
+                                    :description nil,
+                                    :parent-id nil,
+                                    :parent nil}}},
+           :join-tables nil},
+   :fk-field-ids #{},
+   :table-ids #{(id :venues)}}]
+  (let [expanded-form (expand/expand {:database (id)
```

Comment:
I don't actually want them to be refactored at any point and I don't want them to be based on functions that could change over time and modify the query that is built.  I want to describe any MBQL query that could have been written (ideally since the start of the app) and be able to validate that we can still handle them and produce valid expanded/resolved forms.  

So for example, we'd want proper coverage that all of the older MBQL'95 syntax is still resolving as expected, and then going forward with each new addition/modification to the QL we already have a nice body of tests which are validating that everything works.